### PR TITLE
[WIP] Add Blacklight::Elasticsearch::Repository implementation

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by jronallo** [comment]
Is Elasticsearch replacing Solr in Blacklight? It looks like with this pull request at least some generators and templates will default to using ES?

**Comment by mejackreed** [comment]
I hope not.

File: blacklight.gemspec [hunk 1]
```diff
@@ -31,6 +31,9 @@ Gem::Specification.new do |s|
   s.add_dependency "nokogiri",  "~>1.6"     # XML Parser
   s.add_dependency "kaminari", ">= 0.15" # the pagination (page 1,2,3, etc..) of our search results
   s.add_dependency "deprecation"
+  s.add_dependency "elasticsearch-model"
```
Comment (cbeer) [inline blacklight.gemspec]: I wonder if we should be generating these dependencies into downstream apps, instead of requiring them for everyone?

**Comment by jcoyne** [comment]
Yes everyone, please be patient. This is just a rebase of some work @cbeer did 3 years ago. There's plenty of work that needs to be done.

**Pull request closed without merging**